ADR/0011-roles-and-permissions.md

```diff
@@ -0,0 +1,81 @@
+| Contributor   | Workspace               | Access to namespaces that backs workspace                           |
+|               | Application & Component | appstudio.redhat.com      | get, list, watch                        | applications, components, componentdetectionqueries
+|               | Environment             | appstudio.redhat.com      | get, list, watch                        | promotionruns, snapshotenvironmentbindings, snapshots, environments
+|               | *GitOps*                | managed-gitops.redhat.com | get, list, watch                        | gitopsdeployments
+|               | PipelineRun             | tekton.dev                | get, list, watch                        | pipelineruns
+|               | Pipeline Results        | results.tekton.dev        | get, list                               | results, records
+|               | IntegrationTestScenario | appstudio.redhat.com      | get, list, watch                        | integrationtestscenarios
+|               | Enterprise contract     | appstudio.redhat.com      | get, list, watch                        | enterprisecontractpolicies
+|               | Release Strategy        | appstudio.redhat.com      | get, list, watch                        | releases, releasestrategies, releaseplans
+|               | Release Admission Plan  | appstudio.redhat.com      | get, list, watch                        | releaseplanadmissions
+|               | *JVM Build Service*     | jvmbuildservice.io        | get, list, watch                        | jbsconfigs, artifactbuilds
+|               | *Service Access*        | appstudio.redhat.com      | get, list, watch                        | spiaccesstokenbindings, spiaccesschecks, spiaccesstokens, spifilecontentrequest
+|               | *Configs*               |                           | get, list, watch                        | configmaps
+|               | *Secrets*               |                           |                                         | secrets
+|               | Add User                |
+|               | User group (with SSO)   |
+| Maintainer    | Workspace               | Access to namespaces that backs workspace                           |
+|               | Application & Component | appstudio.redhat.com      | get, list, watch, create, update, patch | applications, components, componentdetectionqueries
+|               | Environment             | appstudio.redhat.com      | get, list, watch                        | promotionruns, snapshotenvironmentbindings, snapshots, environments
+|               | *GitOps*                | managed-gitops.redhat.com | get, list, watch                        | gitopsdeployments
+|               | PipelineRun             | tekton.dev                | create, get, list, watch                | pipelineruns
```

Comment:
@sbose78 can you confirm that the end user (maintainer role) should be able to *create* pipelineruns directly? (build-service and pipelines-as-code are typically the initiators, so I'm not sure we _need_ to allow this with our current feature set).

Reply:
I removed the create